include/seastar/core/abort_on_expiry.hh

```diff
@@ -32,6 +33,7 @@ namespace seastar {
 
 /// Facility to tie a timeout with an abort source
 /// Can be used to make abortanle fibers also support timeouts
+SEASTAR_MODULE_EXPORT
```

Comment:
Could we just `export namespace seastar`? This saves us from having to repeat it for every name.

Reply:
export the whole namespace means we are putting *all* declarations and definitions in the namespace into public.  this has following requirements and consequences: 

1. we cannot hide the internal symbols from users. this practically defeats the second goal of introducing Seastar. please see the commit message and cover letter.
2. since all symbols in the namespace have to be exported, they will have external linkage. i'm afraid that this might hurt the performance of linker.

so, i think it's not worthy to go after the "saving" in this area. what i imaging, is that the macro of `SEASTAR_MODULE_EXPORT` is a strong sign that: "a public symbol follows, change it with caution!"

Reply:
> Seastar already has the convention that all public names are exported (but names in seastar::internal are reserved). It's true that some internal names predate the convention and are slowly migrated to seastar::internal, but we can live with it.
> 
> The goal is to reduce noise. Suppose we have module.cc that exports the seastar namespace, then #includes all public headers. Does this help?

not quite. `seastar.cc` includes not only public headers but also private headers. we use the `export` keyword to change the visibility and linkage of a block or a single symbol to external. the ones not marked with `export` are visible only from within the module.

> 
> If not, we can keep the current patch as is.